### PR TITLE
Add Immutable Dispatcher

### LINKED ISSUE #7: Implement ImmutableDispatcher

### PR DESCRIPTION
**Pull request by dcsg**
Closes #7

### CODE REVIEW

File: spec/eventception/immutable_dispatcher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  let(:subscriber) { Eventception::Support::Subscriber.new }
+  let(:subscribers) { [] }
+  let(:event_name) { :on_after }
+  let(:priority) { 0 }
+
+  describe '#initialize' do
+    context 'when one listener is provided' do
+      let(:listeners) {
+        [
+          { event_name: event_name, listener: listener, listener_method: listener_method, priority: priority },
```
Comment (ivoanjo) [inline spec/eventception/immutable_dispatcher_spec.rb]: Minor :nail_care:, there should be no space after `{` and before `}`, as this is a hash, not a block.

File: spec/eventception/immutable_dispatcher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  #     before do
+  #       dispatcher.add_listener(event_name: event_name, listener: listener, listener_method: listener_method)
+  #     end
+  #
+  #     it 'executes the listener method' do
+  #       expect(listener).to receive(listener_method).and_call_original
+  #       expect { dispatcher.dispatch(event_name: event_name) }.to output(/before/).to_stdout
+  #     end
+  #   end
+  # end
```
Comment (ivoanjo) [inline spec/eventception/immutable_dispatcher_spec.rb]: Should this be here...?

File: spec/eventception/immutable_dispatcher_spec.rb (hunk 1 above)
Comment (dcsg) [inline spec/eventception/immutable_dispatcher_spec.rb]: I guess rubocop will complain!

File: spec/eventception/immutable_dispatcher_spec.rb (hunk 2 above)
Comment (dcsg) [inline spec/eventception/immutable_dispatcher_spec.rb]: forgot to delete them, was just to save some work 😅

**Pull request merged**